website/www/site/content/en/documentation/io/built-in/snowflake.md

```diff
@@ -0,0 +1,738 @@
+
+- `--role` Role to use. Optional.
+
+- `--snowPipe` SnowPipe name. Optional.
+
+**Note**: table and query is not in pipeline options by default, it may be added by [extending](https://beam.apache.org/documentation/io/built-in/snowflake/#extending-pipeline-options) PipelineOptions.
+
+Currently, SnowflakeIO **doesn't support** following options at runtime:
+
+- `--url` Snowflake's JDBC-like url including account name and region without any parameters.
+
+- `--oauthToken` Required for OAuth authentication only.
+
+- `--privateKeyPath` Path to Private Key file. Required for Private Key authentication only.
+
+- `--authenticator` Authenticator to use. Optional.
+
+- `--portNumber` Port number. Optional.
+
+- `--loginTimeout` Login timeout. Optional.
+
+## Writing to Snowflake tables
+One of the functions of SnowflakeIO is writing to Snowflake tables. This transformation enables you to finish the Beam pipeline with an output operation that sends the user's [PCollection](https://beam.apache.org/releases/javadoc/2.17.0/org/apache/beam/sdk/values/PCollection.html) to your Snowflake database.
+### Batch write (from a bounded source)
+The basic .`write()` operation usage is as follows:
+{{< highlight java >}}
+data.apply(
+   SnowflakeIO.<type>write()
+       .withDataSourceConfiguration(dc)
+       .toTable("MY_TABLE")
+       .withStagingBucketName("BUCKET NAME")
+       .withStorageIntegrationName("STORAGE INTEGRATION NAME")
+       .withUserDataMapper(mapper)
+)
+{{< /highlight >}}
+Replace type with the data type of the PCollection object to write; for example, SnowflakeIO.<String> for an input PCollection of Strings.
+
+All the below parameters are required:
+
+- `.withDataSourceConfiguration()` Accepts a DatasourceConfiguration object.
+
+- `.toTable()` Accepts the target Snowflake table name.
+
+- `.withStagingBucketName()` Accepts a cloud bucket path ended with slash.
+ -Example: `.withStagingBucketName("gs://mybucket/my/dir/")`
+
+- `.withStorageIntegrationName()` Accepts a name of a Snowflake storage integration object created according to Snowflake documentationt. Example:
+{{< highlight >}}
+CREATE OR REPLACE STORAGE INTEGRATION test_integration 
+TYPE = EXTERNAL_STAGE 
+STORAGE_PROVIDER = GCS 
+ENABLED = TRUE 
+STORAGE_ALLOWED_LOCATIONS = ('gcs://bucket/');
+{{< /highlight >}}
+Then:
+{{< highlight >}}
+.withStorageIntegrationName(test_integration)
+{{< /highlight >}}
+
+- `.withUserDataMapper()` Accepts the UserDataMapper function that will map a user's PCollection to an array of String values `(String[])`.
+
+**Note**:
+SnowflakeIO uses COPY statements behind the scenes to write (using [COPY to table](https://docs.snowflake.net/manuals/sql-reference/sql/copy-into-table.html)). StagingBucketName will be used to save CSV files which will end up in Snowflake. Those CSV files will be saved under the “stagingBucketName” path.
+
+**Optional** for batching:
+- `.withQuotationMark()` 
```

Comment:
I think we should remove it as it is not in Beam yet . @kkucharc ?

Reply:
Agree

Reply:
Done